Library/Homebrew/cmd/install.rb

```diff
@@ -83,12 +77,17 @@ def install
 
       formulae.each { |f| install_formula(f) }
     rescue FormulaUnavailableError => e
-      ofail e.message
-      query = query_regexp(e.name)
-      puts 'Searching formulae...'
-      puts_columns(search_formulae(query))
-      puts 'Searching taps...'
-      puts_columns(search_taps(query))
+      msg = blacklisted? e.name
+      if msg
+        ofail "#{e.message}\n#{msg}"
+      else
```

Comment:
Move blacklisted check here. Because we only show blacklist message when the formula is not available and blacklisted. Right?